bin/main

```diff
@@ -1,72 +1,55 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
-turn = 0
-player = true
-choices = [0]
+require_relative '../lib/board'
+require_relative '../lib/selection_display'
 
-# Check for win and display winner
-def winner
-  puts 'Here we define if a winner is found if it is we display the winner'
-  false
-end
-# Here we end the game either called by winner method or if the turn is > 9
-
-def game_over
-  puts 'Game over'
-end
-puts 'The game is on!'
-puts '-----------------------'
-puts 'Game will start select a valid input for your play (Any number from 0~9)'
+board_controller = Board.new
+local_selector = Selector.new
 
-# Displaying the board
-puts 'Display the game board'
 
-while turn < 9
-  if player
-    puts "Player One's turn"
-    player_one = gets.to_i
-    if player_one < 1 || player_one > 9
-      puts 'Wrong input'
-    else
-      puts "Player's selection displayed one the board"
-      player = false
-      if choices.include? player_one
-        puts 'Choice already taken, Please choose another choice!'
-      else
-        choices << player_one
-        turn += 1
-
-        win = winner
-        if win
-          # Display the player who has won
-          game_over
-        end
-      end
+def init_board(board_c)
+  puts '--------------'
+  board_c.board.each do |n|
+    puts n.join '  |  '
+    puts '--------------'
+  end
+end
 
+def display_player_move(player_mark, board_c, local_s)
+  puts 'Player ' + player_mark + "'s turn, make your pick"
+  puts '--------------'
+  player_pick = gets.to_i
+  if player_pick > 0 && player_pick < 10 && local_s.valid?(player_pick.to_s, board_c.board)
+    board_c.board = local_s.selection(player_mark, player_pick.to_s, board_c.board)
```

Comment:
Sorry, but that the same goal could be reach by moving the needed method into the **Board** class.

1. **local_s.valid?(player_pick.to_s, board_c.board)**
Doing that mean that there is an issue concerning the **Responsibility principle**. Why ? Because, you send as a parameter the board into an external method for it to be use as an **essential element** for checking whether or not the input is valid.
What if the **valid?** method were into the **Board** class ? You will just write
```ruby
if player_pick > 0 && player_pick < 10 && board_c.valid?(local_s)
```
With
```ruby
def valid?(sym)
  # Directly use @board here instead of use it out
end
```

2. **board_c.board = local_s.selection(player_mark, player_pick.to_s, board_c.board)**
Do you really need to modify the board outside by something else and then update if through some kind of setter ?
The simple fact that something else, **outside the Board** class, is modifying the board imply that the board modification is the responsibility of the **Board** class
```ruby
def updateBoard(player_pick)
  # player_mark can be obtained from player_one_turn
  # just re-implement selection method here and update @board attribute
  # ...
  # the code...
end
```

Reply:
Fixed

Reply:
For the first case we are checking for validation on input on a separate class, yes it depends on the board but that doesn't break single responsibility concept since the method does not modify the outsider element.

Please refer here: https://en.wikipedia.org/wiki/Single_responsibility_principle

For the second one you are totally right it breaks the principle.

1 Fixed, 1 Ignored.